src/client/client.c

```diff
@@ -1,32 +1,179 @@
-#include <signal.h>
-#include <stdbool.h>
+#include "client.h"
+#include "libhirte/bus/utils.h"
 #include <stdio.h>
-#include <stdlib.h>
-#include <unistd.h>
 
-// NOLINTNEXTLINE
-static bool done = false;
+int list_units_on_all(UNUSED Client *client) {
+        printf("Listing units on all\n");
+        return 0;
+}
+
+int list_units_on(const char *name, Client *client) {
+        _cleanup_sd_bus_error_ sd_bus_error error = SD_BUS_ERROR_NULL;
+        _cleanup_sd_bus_message_ sd_bus_message *message = NULL;
+
+        int r = 0;
+
+        printf("Listing units on %s\n", name);
+
+        r = assemble_object_path_string(NODE_OBJECT_PATH_PREFIX, name, &client->object_path);
+        if (r < 0) {
+                return r;
+        }
+
+        r = sd_bus_call_method(
+                        client->user_bus,
+                        HIRTE_INTERFACE_BASE_NAME,
+                        client->object_path,
+                        NODE_INTERFACE,
+                        "ListUnits",
+                        &error,
+                        &message,
+                        "");
+        if (r < 0) {
+                fprintf(stderr, "Failed to issue method call: %s\n", error.message);
+                return r;
+        }
+
+        r = sd_bus_message_enter_container(message, SD_BUS_TYPE_ARRAY, "(ssssssouso)");
```

Comment:
Not a blocker, It wouldn't hurt if we had a define for these large type strings. Like UNIT_INFO_TYPESTRING

Reply:
Ack, good point, done